### PR TITLE
feat: support nai-diffusion-3 model

### LINKED ISSUE #212: Feature: 兼容nai-diffusion-3
### Scope

NovelAI

### Describe the problem related to the feature request

nai更新了nai-diffusion-3模型，希望通过插件调用

### Describe the solution you'd like

请求的时候模型参数填nai-diffusion-3即可

### Describe alternatives you've considered

这应该是最简单的吧（

### Additional context

差不多没有

### LINKED ISSUE #215: Feature: 增加NAI采样器的可选项以及采样器SMEA/SMEA DYN选项
### Scope

NovelAI

### Describe the problem related to the feature request

这些是nai官网新增的一些采样器和SMEA/SMEA DYN采样器功能，希望bot能够适配
image
image

### Describe the solution you'd like

image

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #218: Feature: 在参数一致的情况下，机器人生成的图片质量远远低于web端生成的图片。
### Describe the problem related to the feature request

在参数一致的情况下，机器人生成的图片质量远远低于web端生成的图片。
image
这是在机器人上生成的图片

image
这是在web端生成的图片

我最近才使用的novelai，我没有在web端中找到Scale这项设置，倒是只见到这两类似的东西
image
image

### Describe the solution you'd like

{924783DE-5F91-4ac7-9DD8-AD37CE7D855F}
希望更新新的采样接口

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MirrorCY**
fix #212 
fix #215 
fix #218

### CODE REVIEW

**Comment by MirrorCY** [comment]
# V3 Sampler Options

## Samplers
- `k-euler`
- `k_euler_ancestral`
- `k_dpmpp_2s_ancestral`
- `k_dpmpp_2m`
- `k_dpmpp_sde`
- `ddim_v3`

## Notes
- The `"noise_schedule"` is automatically set to `"exponential"` for `k_dpmpp_2m`. For other samplers, it defaults to `"noise_schedule": "native"`.

## Limitations under `ddim_v3` Sampler
- The following features are not supported:
  - `"noise_schedule": "native"` (This option should be removed)
  - `"sm": false`
  - `"sm_dyn": false` (These two options should be set to false)

## New Options
- `"sm": false` (Needs to be enabled for the next option)
- `"sm_dyn": false` (Can be used only when `"sm"` is true)
- `"dynamic_thresholding": bool`

**Comment by MirrorCY** [comment]
A Valid Reques
```
  {
  "input": "1boy, 1girl, blurry background, {red eyes}, medium hair, crown braid, purple hair, gradient hair, blonde hair, blunt ends, hair intakes, flat chest, female service cap, hat ribbon, armor, ok sign, hood, orange eyes, bags under eyes, big hair, asymmetrical hair, brown hair, skinny, multicolored argyle sweater, flip-flops, arm grab, pendant, best quality, amazing quality, very aesthetic, absurdres",
  "model": "nai-diffusion-3",
  "action": "generate",
  "parameters":
    {
      "params_version": 1,
      "width": 1216,
      "height": 2560,
      "scale": 5,
      "sampler": "k_euler_ancestral",
      "steps": 8,
      "n_samples": 1,
      "ucPreset": 0,
      "qualityToggle": true,
      "sm": true,
      "sm_dyn": true,
      "dynamic_thresholding": true,
      "controlnet_strength": 1,
      "legacy": false,
      "add_original_image": false,
      "uncond_scale": 1,
      "cfg_rescale": 0,
      "noise_schedule": "native",
      "legacy_v3_extend": false,
      "seed": 2323460999,
      "negative_prompt": "nsfw, lowres, {bad}, error, fewer, extra, missing, worst quality, jpeg artifacts, bad quality, watermark, unfinished, displeasing, chromatic aberration, signature, extra digits, artistic error, username, scan, [abstract]",
    },
}
  ```

File: src/config.ts [hunk 1]
```diff
@@ -103,9 +112,10 @@ export namespace sampler {
     })).loose().description('默认的采样器。').default('k_euler_a')
   }
 
-  export function sd2nai(sampler: string): string {
+  export function sd2nai(sampler: string, model: string): string {
     if (sampler === 'k_euler_a') return 'k_euler_ancestral'
-    if (sampler in nai) return sampler
+    if (model === 'nai-v3' && sampler in nai3) return sampler
```
Comment (initialencounter) [inline src/config.ts:117]: 这里不应该返回 nai3.sampler 吗

Comment (MirrorCY) [reply src/config.ts:117]: 如果判断正确，说明用户输入了正确的 sampler name

**Comment by MirrorCY** [comment]
@MaikoTan done, take a look?

**Comment by MirrorCY** [comment]
fix #218

**Comment by MirrorCY** [comment]
fix #215

**Comment by MirrorCY** [comment]
fix #212

**Review by shigma — APPROVED** [review]

**Review by MaikoTan — APPROVED** [review]
LGTM

**Pull request merged**